test/stableTRIStaking/transfer.test.ts

```diff
@@ -0,0 +1,179 @@
+import { ethers } from "hardhat";
+import chai from "chai";
+import { solidity } from "ethereum-waffle";
+import { Tri__factory } from "../../typechain";
+
+chai.use(solidity);
+const { expect } = chai;
+
+describe("RevenueDistributionToken - Transfer", function () {
+  beforeEach(async function () {
+    await ethers.provider.send("hardhat_reset", []);
+    //
+    const [deployer, user0, user1] = await ethers.getSigners();
+    this.deployer = deployer;
+    this.user0 = user0;
+    this.user1 = user1;
+
+    const triFactory = new Tri__factory(this.deployer);
+    this.tri = await triFactory.deploy(this.deployer.address);
+    await this.tri.mint(this.deployer.address, "100");
+    this.xTRI = await ethers.getContractFactory("TriBar");
+    this.xTRI = await this.xTRI.deploy(this.tri.address);
+    await this.tri.approve(this.xTRI.address, "100");
+    await this.xTRI.enter("100");
+    this.pTRI = await ethers.getContractFactory("StableTRIStaking");
+    this.ERC20Mock = await ethers.getContractFactory("ERC20Mock");
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    this.rewardToken = await this.ERC20Mock.deploy("rewardToken", "rewardToken", 18, "100");
+    const pTRIConstructorArgs = ["pTRI", "pTRI", this.rewardToken.address, this.tri.address, this.deployer.address, 0];
+    this.pTRI = await this.pTRI.deploy(...pTRIConstructorArgs);
+    await this.tri.approve(this.pTRI.address, "100000000000000000");
+  });
+
+  it("should pre-claim before transfer", async function () {
```

Comment:
can you do the same tests for transferFrom function
also check the value of allowance after the test